app/src/main/java/org/matrix/matrixandroidsdk/car/CarBroadcastReceiver.java

```diff
@@ -0,0 +1,78 @@
+package org.matrix.matrixandroidsdk.car;
+
+import android.app.NotificationManager;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v4.app.RemoteInput;
+import android.util.Log;
+
+import org.matrix.androidsdk.MXSession;
+import org.matrix.androidsdk.data.Room;
+import org.matrix.androidsdk.rest.callback.ApiCallback;
+import org.matrix.androidsdk.rest.model.Event;
+import org.matrix.androidsdk.rest.model.MatrixError;
+import org.matrix.androidsdk.rest.model.Message;
+import org.matrix.matrixandroidsdk.Matrix;
+import org.matrix.matrixandroidsdk.util.NotificationUtils;
+
+public class CarBroadcastReceiver extends BroadcastReceiver {
+    private static final String TAG = CarBroadcastReceiver.class.getSimpleName();
+
+    @Override
+    public void onReceive(final Context context, final Intent intent) {
+        String roomId = intent.getStringExtra(NotificationUtils.EXTRA_ROOM_ID);
+        if (NotificationUtils.ACTION_MESSAGE_HEARD.equals(intent.getAction())) {
+            NotificationManager notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
+            notificationManager.cancelAll();
+        } else if (NotificationUtils.ACTION_MESSAGE_REPLY.equals(intent.getAction())) {
+            final CharSequence reply = getMessageText(intent);
+
+            final MXSession session = Matrix.getInstance(context).getDefaultSession();
+            final Room room = session.getDataHandler().getRoom(roomId);
```

Comment:
``getDefaultSession()`` can return ``null`` according to the docs at https://github.com/matrix-org/matrix-android-sdk/blob/master/app/src/main/java/org/matrix/matrixandroidsdk/Matrix.java#L43 - in which case the app will ``NullPointerException`` here.